Rivet/Functions/Checkpoint-Migration.ps1

```diff
@@ -0,0 +1,46 @@
+
+function Checkpoint-Migration
+{
+    <#
+    .SYNOPSIS
+    Checkpoints the current state of the database so that it can be re-created.
+
+    .DESCRIPTION
+    The `Checkpoint-Migration` function captures the state of a database after all migrations have been applied. The
+    captured state is exported to a `schema.ps1` file that can be applied with Rivet to re-create that state of the
+    database. Before capturing the current state, a Rivet Push is invoked to ensure all migrations have been applied.
+
+    .EXAMPLE
+    Checkpoint-Migration -Database $Database -Environment $Environment -ConfigFilePath $ConfigFilePath
+
+    Demonstrates how to checkpoint a migration.
+    #>
+    [CmdletBinding()]
+    param(
+        # The database(s) to migrate.
+        [Parameter(Mandatory)]
+        [String[]] $Database,
+
+        # The environment you're working in.
+        [String] $Environment,
+
+        # The path to the Rivet configuration file. Default behavior is to look in the current directory for a `rivet.json` file.
+        [String] $ConfigFilePath
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
+
+    [Rivet.Configuration.Configuration]$settings = Get-RivetConfig -Database $Database -Path $ConfigFilePath -Environment $Environment
+    Write-Debug "Checkpoint-Migration: Invoke-Rivet -Push on database(s) ($($settings.Databases.Name -join ', '))"
+    Invoke-Rivet -Push @PSBoundParameters | Write-Debug
+
+    foreach( $databaseItem in $settings.Databases )
+    {
+        Write-Debug "Checkpoint-Migration: Exporting migration on database $($databaseItem.Name)"
+        $migration = Export-Migration -SqlServerName $settings.SqlServerName -Database $databaseItem.Name
+        $migration = $migration -join [Environment]::NewLine
+        $outfilePath = Join-Path -Path $rivetModuleRoot -ChildPath "..\.output\$($databaseItem.Name).ps1"

```

Comment:
The docs say that the migrations are exported to a `schema.ps1` file, which is not what this is doing. Also a couple of other things:

1. Where should the `schema.ps1` file be placed by default? I think maybe by default gets placed next to the `rivet.json` file. We should probably add an `OutputPath` parameter that allows to customize the path to that file.
2. We should check if the `schema.ps1` file exists first and if it does then fail. Add a `Force` switch that allows overwriting an existing `schema.ps1` file.